### PR TITLE
client-go: remove openstack auth plugin

### PR DESCRIPTION
**Pull request by enj**
The openstack auth plugin was deprecated in v1.11 in favor of the "client-keystone-auth" credential plugin.  This change removes the plugin in v1.19 (eight releases seems more than sufficient notice).

Signed-off-by: Monis Khan <mok@vmware.com>

@kubernetes/sig-auth-pr-reviews 
@kubernetes/sig-cli-pr-reviews 
@kubernetes/provider-openstack-pr-reviews 
@dims @hogepodge @dklyle

/kind cleanup
/kind deprecation
/priority important-longterm
/milestone v1.19
/sig auth
/sig cli
/sig openstack
xref: #64346

```release-note
The openstack auth plugin has been removed.
Please use the "client-keystone-auth" kubectl/client-go credential plugin instead.
See https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-client-keystone-auth.md for further details.
```

```
# kubectl configured to use openstack auth-provider
$ kubectl get ns
error: The openstack auth plugin has been removed.
Please use the "client-keystone-auth" kubectl/client-go credential plugin instead.
See https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-client-keystone-auth.md for further details
```

### CODE REVIEW

**Comment by enj** [comment]
/area openstack

?

File: staging/src/k8s.io/client-go/plugin/pkg/client/auth/openstack/openstack_stub.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+func init() {
+	if err := rest.RegisterAuthProviderPlugin("openstack", newOpenstackAuthProvider); err != nil {
+		klog.Fatalf("Failed to register openstack auth plugin: %s", err)
+	}
+}
+
+func newOpenstackAuthProvider(_ string, _ map[string]string, _ rest.AuthProviderConfigPersister) (rest.AuthProvider, error) {
+	return nil, errors.New(`The openstack auth plugin has been removed.
+Please use the "client-keystone-auth" kubectl/client-go credential plugin instead.
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/plugin/pkg/client/auth/openstack/openstack_stub.go]: nice touch

**Comment by liggitt** [comment]
deprecation timeline, mechanics, and replacement stub lgtm.

will defer to @kubernetes/cloud-provider-openstack-maintainers  for lgtm

**Comment by lingxiankong** [comment]
/lgtm

thanks!

**Pull request merged**